R/generate_scenario_info.R

```diff
@@ -57,6 +63,9 @@ ScenarioInfo <- function(# Currently only "Perfect", "Linear", and "Lagged" Expe
   self$mLogitAgroForest <- aLogitAgroForest
   self$mLogitAgroForest_NonPasture <- aLogitAgroForest_NonPasture
   self$mLogitCropland <- aLogitCropland
+  self$mUseZeroCost <- aUseZeroCost
+  self$mCalibrateShareWt <- aCalibrateShareWt
+  self$mShareWeights <- aShareWeights
```

Comment:
We should check to see that if the share weights are not supplied, then we have calibration turned on.
```suggestion
  self$mShareWeights <- aShareWeights
  assertthat::assert_that(!is.null(aShareWeights) || aCalibrateShareWt, msg='If aShareWeights is not supplied, the aCalibrateShareWt must be TRUE.')
```
Arguably it's also a mistake to supply share weights and then turn calibration on, but this won't actually cause an error, so I didn't check for it above.